tests/store/tracking/test_file_store_logged_model.py

```diff
@@ -0,0 +1,646 @@
+import time
+from typing import NamedTuple, Optional
+from unittest import mock
+
+import pytest
+
+from mlflow.entities.lifecycle_stage import LifecycleStage
+from mlflow.entities.logged_model_parameter import LoggedModelParameter
+from mlflow.entities.logged_model_status import LoggedModelStatus
+from mlflow.entities.logged_model_tag import LoggedModelTag
+from mlflow.exceptions import MlflowException
+from mlflow.store.tracking.file_store import SEARCH_LOGGED_MODEL_MAX_RESULTS_DEFAULT, FileStore
+
+
+@pytest.fixture
+def store(tmp_path):
+    return FileStore(str(tmp_path.joinpath("mlruns")))
+
+
+class CreateLoggedModelArgs(NamedTuple):
+    experiment_name: Optional[str] = None
+    run_name: Optional[str] = None
+    model_name: Optional[str] = None
+    tags: Optional[list[LoggedModelTag]] = None
+    params: Optional[list[LoggedModelParameter]] = None
+    model_type: Optional[str] = None
+
+
+@pytest.mark.parametrize(
+    "create_logged_model_args",
+    [
+        CreateLoggedModelArgs(),
+        CreateLoggedModelArgs(experiment_name="test"),
+        CreateLoggedModelArgs(experiment_name="test", run_name="test_run"),
+        CreateLoggedModelArgs(experiment_name="test", run_name="test_run", model_name="test_model"),
+        CreateLoggedModelArgs(
+            experiment_name="test",
+            run_name="test_run",
+            model_name="test_model",
+            tags=[LoggedModelTag("tag_key", "tag_value")],
+        ),
+        CreateLoggedModelArgs(
+            experiment_name="test",
+            run_name="test_run",
+            model_name="test_model",
+            tags=[LoggedModelTag("tag_key", "tag_value")],
+            params=[LoggedModelParameter("param_key", "param_value")],
+        ),
+    ],
+)
+def test_create_logged_model(store, create_logged_model_args):
+    experiment_id = (
+        store.create_experiment(create_logged_model_args.experiment_name)
+        if create_logged_model_args.experiment_name
+        else None
+    )
```

Comment:
I'd avoid `@pytest.mark.parametrize` if you need to insert lots of if statements like this test.

Reply:
Make sense :) Let me update